### DISCUSSION TITLE
Different castling rules for different sides

### DISCUSSION THREAD

**Discussion by pallabbasu** [fairy-stockfish/Fairy-Stockfish#545] [Variants]
I am trying to implement a variant (chess with different armies) where different sides have different castling rules. One side may have a standard FIDE army with rook castling. The other side may have non-standard pieces with no-rooks. However there would be "rook"-like piece (starting in the rook square) which may castle with the king.  

How to do it?

**Comment by ianfab** [comment]
See #52
